control/utils.py

```diff
@@ -175,6 +176,43 @@ def is_valid_nqn(nqn):
 
         return (0, os.strerror(0))
 
+    def get_directory_with_fsid(logger, config, ceph_utils, conf_section, conf_field, def_value, suffix, should_create):
+        status = errno.ENOENT
+        dir_name = config.get_with_default(conf_section, conf_field, "")
+        if dir_name:
+            # If there is an explicit dir name in configuration file, do not change it
+            suffix = ""
+        else:
+            dir_name = def_value
+            if ceph_utils:
+                fsid = ceph_utils.fetch_ceph_fsid()
+                if fsid:
+                    dir_name += fsid + "/"
+        if not dir_name.endswith("/"):
+            dir_name += "/"
+        if suffix:
+            dir_name += suffix
+        if not dir_name.endswith("/"):
+            dir_name += "/"
+        if should_create:
+            try:
+                os.makedirs(dir_name, 0o777, True)
```

Comment:
That's a clear no. Services shouldn't be creating (non-temporary) directories at all. That's a security risk (e.g.: 0x777) and an operational misalignment.

Directories are either created by the package installation scripts (that way when a package is removed, these directories are too), or by the containers + orchestrator (which takes care over on the same responsibility as the package manager).

Reply:
The TLS code doesn't create a directory. The directory creation was before for the SPDK socket file. I just used the same function for both. So, if this a problem is has nothing to do with TLS, but SPDK. Please open an issue and we'll deal with it separately. I try to keep the changes in this PR just for the TLS features.